dashboard/app/controllers/report_abuse_controller.rb

```diff
@@ -19,70 +19,82 @@ def to_honeybadger_context
 class ReportAbuseController < ApplicationController
   AGE_CUSTOM_FIELD_ID = 24_024_923
 
+  # Projects that are created by users with project validator permissions are
+  # blocked from abuse reports because they are created internally and we know
+  # they are safe. This reduces spamming of the report abuse feature.
+  def protected_project?
+    return false if params[:channel_id].blank?
+    owner, _storage_app_id = storage_decrypt_channel_id(params[:channel_id])
+    project_owner = User.find(owner)
+    project_owner.permission?(UserPermission::PROJECT_VALIDATOR)
```

Comment:
nice approach! what have you done to validate that this is working?

hint: I don't think it is possible for this to be working reliably without looking at the user_storage_ids table.

Reply:
🤦‍♀️ Good catch! I updated and added test coverage to verify.